packages/permit/src/funcs.rs

```diff
@@ -2,16 +2,33 @@ use cosmwasm_std::{
     to_binary, Api, Binary, CanonicalAddr, Extern, HumanAddr, Querier, StdError, StdResult, Storage,
 };
 use ripemd160::{Digest, Ripemd160};
-use secp256k1::Secp256k1;
-
+use secp256k1::{PublicKey, Message, ecdsa::Signature as Signature, Secp256k1};
+use sha2::Sha256;
+// use secp256k1::{PublicKey, Message as SecpMessage, Signature, verify};
+use bech32::{ToBase32, Variant};
 use crate::{Permit, RevokedPermits, SignedPermit};
 
+
+pub fn sha_256(data: &[u8]) -> [u8; 32] {
+    let mut hasher = Sha256::new();
+    hasher.update(data);
+    let hash = hasher.finalize();
+
+    let mut result = [0u8; 32];
+    result.copy_from_slice(hash.as_slice());
+    result
+}
```

Comment:
We have this function in `secret_toolkit::crypto`, you could import it from there instead.

Reply:
Oh, I guess you removed it because it didn't link without setting the `rng` feature on 😅